src/Compilers/CSharp/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.cs

```diff
@@ -0,0 +1,418 @@
+        M(3, a: 4);
+    }
+}";
+            var verifier = CompileAndVerify(source, expectedOutput: "First 1 2. Second 3 4.", parseOptions: TestOptions.Regular7_2);
+            verifier.VerifyDiagnostics();
+
+            var tree = verifier.Compilation.SyntaxTrees.First();
+            var model = verifier.Compilation.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var firstInvocation = nodes.OfType<InvocationExpressionSyntax>().ElementAt(2);
+            Assert.Equal("M(a: 1, 2)", firstInvocation.ToString());
+            Assert.Equal("void C.M(System.Int32 a, System.Int32 b)",
+                model.GetSymbolInfo(firstInvocation).Symbol.ToTestDisplayString());
+
+            var secondInvocation = nodes.OfType<InvocationExpressionSyntax>().ElementAt(3);
+            Assert.Equal("M(3, a: 4)", secondInvocation.ToString());
+            Assert.Equal("void C.M(System.Int64 b, System.Int64 a)",
+                model.GetSymbolInfo(secondInvocation).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void TestPositionalUnaffected()
+        {
+            var source = @"
+class C
+{
+    static void M(int first, int other)
+    {
+        System.Console.Write($""{first} {other}"");
+    }
+    static void Main()
+    {
+        M(1, first: 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (10,14): error CS1744: Named argument 'first' specifies a parameter for which a positional argument has already been given
+                //         M(1, first: 2);
+                Diagnostic(ErrorCode.ERR_NamedArgumentUsedInPositional, "first").WithArguments("first").WithLocation(10, 14)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<InvocationExpressionSyntax>().ElementAt(1);
+            Assert.Equal("M(1, first: 2)", invocation.ToString());
+            Assert.Null(model.GetSymbolInfo(invocation).Symbol);
+        }
+
+        [Fact]
+        public void TestPositionalUnaffected2()
+        {
+            var source = @"
+class C
+{
+    static void M(int a, int b, int c = 1)
+    {
+        System.Console.Write($""M {a} {b}"");
+    }
+    static void Main()
+    {
+        M(c: 1, 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (10,9): error CS7036: There is no argument given that corresponds to the required formal parameter 'a' of 'C.M(int, int, int)'
+                //         M(c: 1, 2);
+                Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "M").WithArguments("a", "C.M(int, int, int)").WithLocation(10, 9)
```

Comment:
I think it will be difficult to produce two errors. AnalyzeArguments returns a single ArgumentAnalysisResult, which typically points to a single argument or parameter with a problem.

---
In reply to: [122122322](https://github.com/dotnet/roslyn/pull/20121#discussion_r122122322) [](ancestors = 122122322)